src/isd/record.py

```diff
@@ -137,38 +142,86 @@ def parse(cls, line: str) -> "Record":
             original_observation_data=original_observation_data,
         )
 
-    def datetime(self) -> datetime.datetime:
+    def timestamp(self) -> datetime.datetime:
         """Returns this record's datetime."""
         return datetime.datetime(
             self.year, self.month, self.day, self.hour, self.minute
         )
 
+    @staticmethod
+    def _extract_data(message: str, tag: str, later_tags: List[str]) -> Tuple[str, str]:
+        if message.startswith(tag):
+            index = None
+            for other_tag in later_tags:
+                try:
+                    index = message.find(other_tag)
+                except ValueError:
+                    continue
+                break
+            if index != -1:
+                data = message[len(tag) : index]
+                tail = message[index:]
+                return data, tail
+            else:
+                return message[len(tag) :], ""
+        else:
+            return "", message
 
-def extract_data(message: str, tag: str, later_tags: List[str]) -> Tuple[str, str]:
-    if message.startswith(tag):
-        index = None
-        for other_tag in later_tags:
-            try:
-                index = message.find(other_tag)
-            except ValueError:
-                continue
-            break
-        if index != -1:
-            data = message[len(tag) : index]
-            tail = message[index:]
-            return data, tail
+    @staticmethod
+    def optional(
```

Comment:
This wasn't "private" before, but probably should be.

```suggestion
    def _optional(
```